File: src/main/java/io/github/mac9p/shopapp/Model/OrderItem.java
package io.github.mac9p.shopapp.Model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import org.hibernate.validator.constraints.URL;

import javax.persistence.*;
import javax.validation.constraints.PositiveOrZero;

@Entity
@Data
public class OrderItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @URL
    private String imageUrl;
    @PositiveOrZero
    private Double price;
    @PositiveOrZero
    private Integer quantity;
    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;
    private Long productId;
}

File: frontend/angular-shopapp/src/app/common/order-item.ts
import {CartItem} from "./cart-item";

export class OrderItem {
  private imageUrl!: string;
  private quantity!: number;
  private unitPrice!: number;
  private productId!: number;

  constructor(cartItem: CartItem) {
    this.imageUrl = cartItem.imageUrl;
    this.quantity = cartItem.quantity;
    this.unitPrice = cartItem.unitPrice;
    this.productId = cartItem.id
  }
}
